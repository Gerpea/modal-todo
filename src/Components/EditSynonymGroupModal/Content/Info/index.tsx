import React from 'react';

import Icon, { Icons } from 'Components/Icon';

import style from './styles.module.scss';

function Info(): React.ReactElement {
  return (
    <div className={style.Info}>
      <h4>Синонимы</h4>
      <Icon color="gray-3">{Icons.info}</Icon>
    </div>
  );
}

export default Info;
