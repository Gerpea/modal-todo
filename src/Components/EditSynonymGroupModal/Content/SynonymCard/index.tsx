import React from 'react';

import { Synonym } from 'Types';

import IconButton from 'Components/IconButton';
import { Icons } from 'Components/Icon';

import style from './styles.module.scss';

export interface Props {
  children: Synonym;
  onEdit: (synonym: Synonym) => void;
  onDelete: (synonym: Synonym) => void;
}

function SynonymCard({ onEdit, onDelete, children: synonym }: Props): React.ReactElement {
  return (
    <div className={style.SynonymCard}>
      <p className={style.SynonymCardText}>{synonym.word}</p>
      <div className={style.SynonymCardControls}>
        <IconButton
          color="blue"
          onClick={() => {
            onEdit(synonym);
          }}
        >
          {Icons.pencil}
        </IconButton>
        <IconButton
          color="red"
          onClick={() => {
            onDelete(synonym);
          }}
        >
          {Icons.trash}
        </IconButton>
      </div>
    </div>
  );
}

export default SynonymCard;
