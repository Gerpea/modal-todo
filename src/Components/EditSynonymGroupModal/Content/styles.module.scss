.Content {
  display: flex;
  flex-direction: column;
  row-gap: 5px;
}

.AddPadding {
  padding-bottom: 5px;
}
