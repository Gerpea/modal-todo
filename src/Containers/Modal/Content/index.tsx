import React from 'react';

import style from '../styles.module.css';

export interface Props {
  children: React.ReactNode;
}

function ModalContent({ children }: Props): React.ReactElement {
  return <div className={style.Content}>{children}</div>;
}

export default ModalContent;
