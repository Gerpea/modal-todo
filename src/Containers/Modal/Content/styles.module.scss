.Content {
  overflow: auto;
  height: 100%;
}
