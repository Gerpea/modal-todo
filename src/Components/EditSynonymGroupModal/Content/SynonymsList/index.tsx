import React from 'react';

import { Id, Synonym, SynonymsState } from 'Types';

import SynonymCard from '../SynonymCard';
import EditSynonym from '../SynonymInputs/EditSynonym';

import style from './styles.module.scss';

export interface Props {
  synonyms: SynonymsState;
  editedId: Id | undefined;
  onUpdate: (synonym: Synonym) => void;
  onEdit: (synonym: Synonym) => void;
  onCancelEdit: () => void;
  onRemove: (synonym: Synonym) => void;
}

function SynonymsList({
  synonyms,
  onUpdate,
  onEdit,
  onCancelEdit,
  onRemove,
  editedId,
}: Props): React.ReactElement {
  return (
    <ul className={style.List}>
      {synonyms.map((synonym) =>
        editedId === synonym.id ? (
          <EditSynonym
            className={style.EditPadding}
            initialValue={synonym.word}
            onEdit={(value) => onUpdate({ id: editedId, word: value })}
            onCancelEdit={onCancelEdit}
            key={synonym.id}
          />
        ) : (
          <SynonymCard onEdit={onEdit} onDelete={onRemove} key={synonym.id}>
            {synonym}
          </SynonymCard>
        ),
      )}
    </ul>
  );
}

export default SynonymsList;
