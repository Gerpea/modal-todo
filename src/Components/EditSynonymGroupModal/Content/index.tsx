import React, { useState } from 'react';

import { Id, Synonym, Word } from 'Types';

import { useAppDispatch } from 'Hooks/useAppDispatch';

import useSynonymsSelector from 'Store/Selectors/synonyms';
import { createSynonym, removeSynonym, updateSynonym } from 'Store/Slices/synonyms';

import AddSynonym from './SynonymInputs/AddSynonym';
import Info from './Info';
import SynonymsList from './SynonymsList';

import style from './styles.module.scss';

function Content(): React.ReactElement {
  const synonyms = useSynonymsSelector();
  const dispatch = useAppDispatch();

  const [editedId, setEditedId] = useState<Id>();

  const handleStartEditSynonym = (synonym: Synonym): void => {
    setEditedId(synonym.id);
  };

  const handleCancelEditSynonym = (): void => {
    setEditedId(undefined);
  };

  const handleRemoveSynonyms = (synonym: Synonym): void => {
    dispatch(removeSynonym(synonym.id));
  };

  const handleUpdateSynonym = (synonym: Synonym): void => {
    if (editedId) {
      dispatch(updateSynonym({ id: editedId, word: synonym.word }));
      setEditedId(undefined);
    }
  };

  const handleAddSynonym = (synonym: Word): void => {
    dispatch(createSynonym(synonym));
  };

  return (
    <div className={style.Content}>
      <Info />
      {!editedId && <AddSynonym onAddSynonym={handleAddSynonym} className={style.AddPadding} />}
      <SynonymsList
        synonyms={synonyms}
        editedId={editedId}
        onEdit={handleStartEditSynonym}
        onCancelEdit={handleCancelEditSynonym}
        onRemove={handleRemoveSynonyms}
        onUpdate={handleUpdateSynonym}
      />
    </div>
  );
}

export default Content;
