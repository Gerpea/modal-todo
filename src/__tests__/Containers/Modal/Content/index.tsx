import React from 'react';
import renderer from 'react-test-renderer';

import Content from 'Containers/Modal/Content';

describe('ModalContent test', () => {
  it('Content renders', () => {
    const tree = renderer.create(<Content>content</Content>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
