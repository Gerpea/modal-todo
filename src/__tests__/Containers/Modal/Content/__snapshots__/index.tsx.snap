// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalContent test Content renders 1`] = `
<div
  className="Content"
>
  content
</div>
`;
